rust/src/transformations/cast_metric/traits.rs

```diff
@@ -6,7 +6,16 @@ use crate::{
     transformations::DatasetMetric,
 };
 
+/// UnorderedMetric is implemented for metrics that compute the distance between datasets in a way that is insensitive to row ordering.
+///
+/// # Proof Definition
+/// For any choice of x and x' where x' = shuffle(x), then $d(x, x') = 0$.
 pub trait UnorderedMetric: DatasetMetric<Distance = IntDistance> {
+    /// OrderedMetric denotes the associated metric that is sensitive to row ordering.
+    ///
+    /// # Proof Definition
+    /// For any choice of dataset x, x' where d(x, x') > 0 wrt OrderedMetric and only differs by the reordering of rows,
+    /// then d(x, x') = 0 wrt UnorderedMetric.
```

Comment:
All the statements of this form are a little confusing because `d` is two different things. Maybe something like:
```suggestion
    /// For any choice of datasets x and x' where d(x, x') > 0 for an OrderedMetric d, and x and x' only differ by the reordering of rows,
    /// then d'(x, x') = 0 for a corresponding UnorderedMetric d'.
```